src/dune_fmt.ml

```diff
@@ -0,0 +1,106 @@
+open! Import
+
+let parse_file path_opt =
+  let fname, contents =
+    match path_opt with
+    | Some path ->
+      Io.with_file_in path ~f:(fun ic ->
+        let contents = Io.read_all ic in
+        (Path.to_string path, contents)
+      )
+    | None ->
+      let lines = Io.input_lines stdin in
+      let contents = String.concat ~sep:"" lines in
```

Comment:
`sep` should be `"\n"`, otherwise 2 lines that contain a single atom would end up concatenated as a single one.